diff --git a/dist/core.mjs b/dist/core.mjs
index 6d5b20ddd9f4ee268f58b049520237ebca3b2682..350f819d154dfe7834aada6d849d91011d951786 100644
--- a/dist/core.mjs
+++ b/dist/core.mjs
@@ -1,4 +1,5 @@
 import { addClassToHast } from 'shikiji-core';
+import { fromHtml } from 'hast-util-from-html';
 
 const module = {
 	type: "element",
@@ -390,15 +391,14 @@ function rendererRich(options = {}) {
     if (jsdoc && info.docs) {
       const docs = processHoverDocs(info.docs) ?? info.docs;
       if (docs) {
+        const hast = fromHtml(docs, { space: 'p', fragment: true }).children[0];
+
         themedContent.push({
           type: "element",
           tagName: "div",
           properties: { class: "twoslash-popup-jsdoc" },
           children: [
-            {
-              type: "text",
-              value: docs
-            }
+            hast
           ]
         });
       }
